File: Third Year/Semester 5/Mobile Applications/Laboratory 2/Meow/Meow/Utilities/CatBreed.swift
import SwiftUI

enum CatBreed: String, CaseIterable, Identifiable {
    case none = "None"
    case abyssinian = "Abyssinian"
    case americanBobtail = "American Bobtail"
    case americanCurl = "American Curl"
    case americanShorthair = "American Shorthair"
    case americanWirehair = "American Wirehair"
    case australianMist = "Australian Mist"
    case balinese = "Balinese"
    case bengal = "Bengal"
    case birman = "Birman"
    case bombay = "Bombay"
    case britishShorthair = "British Shorthair"
    case californiaSpangled = "California Spangled"
    case carey = "Carey"
    case cornishRex = "Cornish Rex"
    case britishLonghair = "British Longhair"
    case devonRex = "Devon Rex"
    case egyptianMau = "Egyptian Mau"
    case gingerTabby = "Ginger Tabby"
    case havanaBrown = "Havana Brown"
    case khaoManee = "Khao Manee"
    case maineCoon = "Maine Coon"
    case multicolor = "Multicolor"
    case munchkin = "Munchkin"
    case norwegianForest = "Norwegian Forest"
    case orientalShorthair = "Oriental Shorthair"
    case persian = "Persian"
    case ragdoll = "Ragdoll"
    case russianBlue = "Russian Blue"
    case scottishFold = "Scottish Fold"
    case siamese = "Siamese"
    case somali = "Somali"
    case sphynx = "Sphynx"
    case tonkinese = "Tonkinese"
    case toyger = "Toyger"
    case tuxedo = "Tuxedo"
    
    var image: Image {
        Image(self.rawValue)
    }
    
    var id: String { self.rawValue }
}

File: Third Year/Semester 5/Mobile Applications/Laboratory 3/Meow/lib/Utilities/CatBreed.dart
enum CatBreed {
  None,
  Abyssinian,
  AmericanShorthair,
  Bengal,
  Birman,
  Bombay,
  BritishShorthair,
  DevonRex,
  GingerTabby,
  MaineCoon,
  Munchkin,
  NorwegianForest,
  OrientalShorthair,
  Persian,
  Ragdoll,
  RussianBlue,
  ScottishFold,
  Siamese,
  Spynx,
  Toyger,
}

extension CatBreedExtension on CatBreed {
  String get stringValue {
    switch (this) {
      case CatBreed.None:
        return "None";
      case CatBreed.Abyssinian:
        return "Abyssinian";
      case CatBreed.AmericanShorthair:
        return "American Shorthair";
      case CatBreed.Bengal:
        return "Bengal";
      case CatBreed.Birman:
        return "Birman";
      case CatBreed.Bombay:
        return "Bombay";
      case CatBreed.BritishShorthair:
        return "British Shorthair";
      case CatBreed.DevonRex:
        return "Devon Rex";
      case CatBreed.GingerTabby:
        return "Ginger Tabby";
      case CatBreed.MaineCoon:
        return "Maine Coon";
      case CatBreed.Munchkin:
        return "Munchkin";
      case CatBreed.NorwegianForest:
        return "Norwegian Forest";
      case CatBreed.OrientalShorthair:
        return "Oriental Shorthair";
      case CatBreed.Persian:
        return "Persian";
      case CatBreed.Ragdoll:
        return "Ragdoll";
      case CatBreed.RussianBlue:
        return "Russian Blue";
      case CatBreed.ScottishFold:
        return "Scottish Fold";
      case CatBreed.Siamese:
        return "Siamese";
      case CatBreed.Spynx:
        return "Sphynx";
      case CatBreed.Toyger:
        return "Toyger";
    }
  }
}